src/psyclone/psyir/backend/sympy_writer.py

```diff
@@ -599,7 +604,10 @@ def reference_node(self, node):
         # Support renaming a symbol (e.g. if it is a reserved Python name).
         # Look up with the name as tag, which will return the symbol with
         # a unique name (e.g. lambda --> lambda_1):
-        name = self._symbol_table.lookup_with_tag(node.name).name
+        try:
+            name = self._symbol_table.lookup_with_tag(node.name).name
+        except KeyError:
+            name = node.name
```

Comment:
I believe only the symbols that need renaming (because they clash with a reserved name, are stored with a tag), if no tag is found this name does not need to be renamed. I added a comment.